src/ui/Facet/FacetSearchDropdownNavigation/FacetSearchDropdownNavigator.ts

```diff
@@ -1,98 +1,78 @@
+
+    if (this.willExcludeCurrentResult) {
       this.announceCurrentResultCanBeExcluded();
+      return;
     }
+
+    this.defaultDropdownNavigator.moveCurrentResultDown();
   }
 
   public focusPreviousElement() {
-    if (!this.canExcludeCurrentResult) {
-      this.moveResultUpAndAnnounce();
-    } else {
+    if (this.willExcludeCurrentResult) {
       this.toggleCanExcludeCurrentResult();
-      this.announceCurrentResultCanBeSelected();
+      return;
     }
-  }
 
-  private moveResultDownAndAnnounce() {
-    this.defaultDropdownNavigator.moveCurrentResultDown();
-    this.announceCurrentResultCanBeSelected();
+    this.moveResultUp();
+    this.toggleCanExcludeCurrentResult();
   }
 
-  private moveResultUpAndAnnounce() {
+  private moveResultUp() {
     this.defaultDropdownNavigator.moveCurrentResultUp();
 
-    if (this.isCurrentResultSelectAllButton) {
-      this.announceCurrentResultCanBeSelected();
-    } else {
+    if (this.willExcludeCurrentResult) {
       this.toggleCanExcludeCurrentResult();
       this.announceCurrentResultCanBeExcluded();
     }
   }
 
-  private get isCurrentResultSelectAllButton() {
-    return this.currentResult.hasClass('coveo-facet-search-select-all');
+  private get isCurrentResultNotAFacetValue() {
+    return this.currentResult.hasClass('coveo-facet-search-select-all') || this.currentResult.hasClass('coveo-facet-value-not-found');
   }
 
-  private get canExcludeCurrentResult() {
+  private get willExcludeCurrentResult() {
     return this.currentResult.hasClass('coveo-facet-value-will-exclude');
   }
 
   private toggleCanExcludeCurrentResult() {
-    this.currentResult.toggleClass('coveo-facet-value-will-exclude', !this.canExcludeCurrentResult);
-  }
+    if (this.isCurrentResultNotAFacetValue) {
+      return;
+    }
 
-  private announceCurrentResultAction() {
-    this.canExcludeCurrentResult ? this.announceCurrentResultCanBeExcluded() : this.announceCurrentResultCanBeSelected();
+    this.currentResult.toggleClass('coveo-facet-value-will-exclude', !this.willExcludeCurrentResult);
   }
 
   private announceCurrentResultCanBeExcluded() {
-    const excludeIconTitle = $$(this.currentResult).find('.coveo-facet-value-exclude').title;
-    this.config.facetSearch.updateAriaLive(excludeIconTitle);
-  }
-
-  private announceCurrentResultCanBeSelected() {
-    const message = this.currentResultSelectMessage;
-    this.config.facetSearch.updateAriaLive(message);
-  }
-
-  private get currentResultSelectMessage() {
-    if (this.isCurrentResultSelectAllButton) {
-      return this.currentResult.text();
+    if (this.isCurrentResultNotAFacetValue) {
+      return;
     }
 
-    const checkbox = this.currentResult.find('.coveo-facet-value-checkbox');
-    return checkbox.getAttribute('aria-label');
+    const excludeIcon = $$(this.currentResult).find('.coveo-facet-value-exclude');
+    this.config.facetSearch.updateAriaLive(excludeIcon.getAttribute('aria-label'));
```

Comment:
Refactored FacetSearchDropdownNavigator.ts  to only call the aria live when the exclude is made available to the user when he uses the arrows. The aria-label does a good job on the elements to notify the select on other occasions.

Also made the choice to only offer the exclude when the user goes down on the facet. I found it simplified the code and made going back up in the results less tedious for the user, let me know what you think.

Reply:
I think this is a nice improvement 👍